### DISCUSSION TITLE
Will this work on a vanilla Ubuntu/Debian?

### DISCUSSION THREAD

**Discussion by Kailee71** [Jip-Hop/jailmaker#97] [Q&A]
I'm really appreciating the simplicity and reliability of Jailmaker. I would actually prefer using this above proxmox in some use cases - with a vanilla Ubuntu or Debian underneath.
Will Jailmaker also work like that or only on a base TrueNAS install?

**Comment by Jip-Hop** [comment]
Wouldn't you rather use systemd-nspawn directly in those cases? Anyhow glad to hear you appreciate jailmaker. I think it will work fine there also. You may get a warning about it not being stored under `/mnt` depending on where you store the script. But keep in mind jailmaker is not tested by me on those vanilla distros.

**Comment by Kailee71** [comment]
It actually worked just fine, with just one exception; gpu passthrough. I get this when I start a jail with nvidia gpu passthrough:

`root@gpunode0:~# jlmkr start ubuntu1
modprobe: FATAL: Module nvidia-current-uvm not found in directory /lib/modules/5.15.0-97-generic

Failed to load nvidia-current-uvm kernel module.
Skip passthrough of nvidia GPU.

Starting jail ubuntu1 with the following command:

systemd-run --collect --property=Delegate=yes --property=RestartForceExitStatus=133 --property=SuccessExitStatus=133 --property=TasksMax=infinity --property=Type=notify --setenv=SYSTEMD_NSPAWN_LOCK=0 --property=KillMode=mixed --unit=jlmkr-ubuntu1 --working-directory=./jails/ubuntu1 '--description=My nspawn jail ubuntu1 [created with jailmaker]' -- systemd-nspawn --bind-ro=/sys/module --boot --inaccessible=/sys/module/apparmor --quiet --keep-unit --machine=ubuntu1 --directory=rootfs '--property=DeviceAllow=char-drm rw' --network-bridge=br0 --resolv-conf=bind-host

Running as unit: jlmkr-ubuntu1.service
root@gpunode0:~# `

Driver (535-server) and cuda-toolkit are installed and fundtional on the host. Do I need to do the bind-mounts manually? How do I check?

**Comment by Jip-Hop** [comment]
Maybe you can comment out the part which tries to load the nvidia module as a test? See if it gets further. I suppose this kernel module is already loaded on Ubuntu so shouldn't be needed. I think you also need to install the nvidia container toolkit though.

**Reply by Jip-Hop** [reply]
Please try again with jlmkr.py from the develop branch. And ensure you have the `nvidia-container-cli` command available locally (part of the NVIDIA Container Toolkit).
